### PR TITLE
Rss acc and viewer

### PR DESCRIPTION
**Pull request by klemense1**
- Bugfix AppendLinesNoIntersect 
- Some Python Wrapping
- Extension for Runtime to extract time series of simulation
- Making shape in LaneCorridorConfig settable using radius and wheelbase
- Fixing LastAction for IDMLaneTracking
- Bugfix ValidAtTime Check in Execution 
- Acceleration Restrictions to Safety Model BehaviorRSSConformant

### CODE REVIEW

File: bark/models/behavior/idm/idm_lane_tracking.hpp [hunk 1]
```diff
@@ -32,6 +32,9 @@ class BehaviorIDMLaneTracking : public BaseIDM {
     crosstrack_error_gain_ =
         params->GetReal("BehaviorIDMLaneTracking::CrosstrackErrorGain",
                         "Tuning factor of stanley controller", 1.0);
+    dynamic::Input input(2);
+    input << 0.0, 0.0;
+    SetLastAction(input);
```
Comment (patrickhart) [inline bark/models/behavior/idm/idm_lane_tracking.hpp:37]: why is this required in the constructor?

File: bark/world/world.cpp [hunk 2]
```diff
@@ -72,7 +72,7 @@ void World::Execute(const double& delta_time) {
   const double inc_world_time = world_time_ + delta_time;
   using models::dynamic::StateDefinition::TIME_POSITION;
   for (auto agent : agents_) {
-    if (agent.second->IsValidAtTime(inc_world_time) &&
+    if (agent.second->IsValidAtTime(world_time_) &&
```
Comment (patrickhart) [inline bark/world/world.cpp:75]: what does this change do? was it not working before?

**Review by patrickhart — APPROVED** [review]

File: bark/world/world.cpp (hunk 2 above)
Comment (klemense1) [reply bark/world/world.cpp:75]: that's only relevant when simulating the interaction dataset in closed loop. It's an error which I introduced when refactoring the Execute function. With this change now, it's the same as it was before.

File: bark/models/behavior/idm/idm_lane_tracking.hpp (hunk 1 above)
Comment (klemense1) [reply bark/models/behavior/idm/idm_lane_tracking.hpp:37]: Cause this https://github.com/bark-simulator/bark/blob/master/bark/models/behavior/idm/base_idm.cpp#L78 needs to be overwritten after the  BaseIDM constructor is called in the BehaviorIDMLaneTracking constructor

File: bark/examples/merging.py [hunk 3]
```diff
@@ -109,5 +110,8 @@ def goal(self, world):
   for step in range(0, 20):
     env.step()
     time.sleep(sim_step_time/sim_real_time_factor)
+    
+  df = env.ExtractTimeSeries()
+  print(df)
```
Comment (tobiaskessler) [inline bark/examples/merging.py:115]: remove print?

File: bark/models/behavior/behavior_rss/behavior_rss.cpp [hunk 4]
```diff
@@ -103,20 +114,36 @@ Trajectory BehaviorRSSConformant::Plan(
...
+
+void BehaviorRSSConformant::ApplyRestrictionsToSafetyModel(
+    const AccelerationLimits& limits) {
+  VLOG(4) << "AccelerationLimits for safety model " << limits;
+  std::shared_ptr<BehaviorIDMLaneTracking> safety_behavior_ptr =
+      std::dynamic_pointer_cast<BehaviorIDMLaneTracking>(
+          behavior_safety_model_);
+  safety_behavior_ptr->SetAccelerationLimits(limits);
+}
+
```
Comment (tobiaskessler) [inline bark/models/behavior/behavior_rss/behavior_rss.cpp:146]: I think it is ok to have ApplyRestrictionsToNominalModel and ApplyRestrictionsToSafetyModel. But it could be one template function

**Review by tobiaskessler — APPROVED** [review]
looks good!

File: bark/examples/merging.py (hunk 3 above)
Comment (klemense1) [reply bark/examples/merging.py:115]: I guess for an example that's sufficient.
